src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/OptionDefinition.cs

```diff
@@ -49,6 +50,15 @@ public OptionDefinition(OptionGroup? group, string configName, object? defaultVa
             DefaultValue = defaultValue;
         }
 
+        public static string? GetConfigNameLanguagePrefix(string languageName)
+            => languageName switch
+            {
+                LanguageNames.CSharp => CSharpConfigNamePrefix,
+                LanguageNames.VisualBasic => VisualBasicConfigNamePrefix,
+                LanguageNames.FSharp => FSharpConfigNamePrefix,
+                _ => null
+            };
```

Comment:
Is this consumed via IVT? Consider adding a comment for that, so that it doesn't get unintentionally deleted as unused in the future.

I'm also wondering why would VS4Mac need to have this? and if it needs it, why it's implemented here and not in VS4Mac itself?

Reply:
Yes, used via IVT. I'll add comment.

This is used to construct language-specific config name from language agnostic name for options that are per-language.

`csharp_dotnet_style_readonly_field` for C# value of `dotnet_style_readonly_field` option.
This name would be used in VS Mac configuration file.

Wondering if this is the best naming for these, or if something like `dotnet_style_readonly_field:csharp` would be better.

@jmarolf Any opinions?

Reply:
> Yes, used via IVT. I'll add comment.

@tmat Isn't it better for such method to be implemented in VS4Mac rather than here and exposed via IVT? The method is really small and doesn't depend on any Roslyn internals.

> E.g. `csharp_dotnet_style_readonly_field` for C# value of `dotnet_style_readonly_field` option.

I'm not sure how this works, but clearly `csharp_dotnet_style_readonly_field` isn't an editorconfig key right? where does it come from? The editorconfig key is `dotnet_style_readonly_field`. Does it get prefixed with `csharp_` somewhere? Is it to differentiate different per-language values?

---

PS: In general, I'd very much appreciate if you follow-up with a markdown doc file after you complete all the options refactoring work.

Reply:
> I'm not sure how this works, but clearly csharp_dotnet_style_readonly_field isn't an editorconfig key right? where does it come from? The editorconfig key is dotnet_style_readonly_field. Does it get prefixed with csharp_ somewhere? Is it to differentiate different per-language values?

Yes, `csharp_dotnet_style_readonly_field` would be used in VS Mac XML setting file to represent the value of `dotnet_style_readonly_field` for C#.